src/fsharp/DependencyManager.Integration.fs

```diff
@@ -0,0 +1,182 @@
+            if isNull theMethod then
+                None
+            else
+                Some theMethod
+        with | _ -> None
+
+    let stripTieWrapper (e:Exception) =
+        match e with
+        | :? TargetInvocationException as e->
+            e.InnerException
+        | _ -> e
+
+(* Shape of Dependency Manager contract, resolved using reflection *)
+type internal IDependencyManagerProvider =
+    abstract Name : string
+    abstract Key: string
+    abstract ResolveDependencies : scriptDir: string * mainScriptName: string * scriptName: string * packageManagerTextLines: string seq * tfm: string -> bool * string list * string list
+
+[<RequireQualifiedAccess>]
+type ReferenceType =
+| RegisteredDependencyManager of IDependencyManagerProvider
+| Library of string
+| UnknownType
+
+type ReflectionDependencyManagerProvider(theType: Type, nameProperty: PropertyInfo, keyProperty: PropertyInfo, resolveDeps: MethodInfo, outputDir: string option) =
+    let instance = Activator.CreateInstance(theType, [|outputDir :> obj|])
+    let nameProperty     = nameProperty.GetValue >> string
+    let keyProperty      = keyProperty.GetValue >> string
+
+    static member InstanceMaker (theType: System.Type, outputDir: string option) =
+        match ReflectionHelper.getAttributeNamed theType dependencyManagerAttributeName,
+                ReflectionHelper.getInstanceProperty<string> theType namePropertyName,
+                ReflectionHelper.getInstanceProperty<string> theType keyPropertyName,
+                ReflectionHelper.getInstanceMethod<bool * string list * string list> theType [| typeof<string>; typeof<string>; typeof<string>; typeof<string seq>; typeof<string> |] resolveDependenciesMethodName
+                with
+        | None, _, _, _ -> None
+        | _, None, _, _ -> None
+        | _, _, None, _ -> None
+        | _, _, _, None -> None
+        | Some _, Some nameProperty, Some keyProperty, Some resolveDependenciesMethod ->
+            Some (fun () -> new ReflectionDependencyManagerProvider(theType, nameProperty, keyProperty, resolveDependenciesMethod, outputDir) :> IDependencyManagerProvider)
+
+    interface IDependencyManagerProvider with
+        member __.Name     = instance |> nameProperty
+        member __.Key      = instance |> keyProperty
+        member __.ResolveDependencies(scriptDir, mainScriptName, scriptName, packageManagerTextLines, tfm) =
+            let arguments = [| box scriptDir; box mainScriptName; box scriptName; box packageManagerTextLines; box tfm |]
+            resolveDeps.Invoke(instance, arguments) :?> _
+
+// Resolution Path = Location of FSharp.Compiler.Private.dll
+let assemblySearchPaths = lazy (
+    [
+        let assemblyLocation = typeof<IDependencyManagerProvider>.GetTypeInfo().Assembly.Location
+        yield Path.GetDirectoryName assemblyLocation
+        yield AppDomain.CurrentDomain.BaseDirectory
+    ])
+
+let enumerateDependencyManagerAssemblies compilerTools m =
```

Comment:
Is this going to conform to the RFC https://github.com/fsharp/fslang-design/blob/master/tooling/FST-1027-fsi-references.md#handler-resolution or does the RFC need to be adjusted?